code/modules/mob/living/silicon/robot/robot_hud.dm

```diff
@@ -0,0 +1,131 @@
+	if(disabilities & NEARSIGHTED)
+		overlay_fullscreen("impaired", /obj/abstract/screen/fullscreen/impaired)
+	else
+		clear_fullscreen("impaired")
+	if(eye_blurry)
+		overlay_fullscreen("blurry", /obj/abstract/screen/fullscreen/blurry)
+	else
+		clear_fullscreen("blurry")
+	if(druggy)
+		overlay_fullscreen("high", /obj/abstract/screen/fullscreen/high)
+	else
+		clear_fullscreen("high")
+
+	if(!isDead())
+		if(machine)
+			if(!(machine.check_eye(src)))
+				reset_view(null)
+		else
+			if(client && !client.adminobs && !iscamera(client.eye) && !isTeleViewing(client.eye))
+				reset_view(null)
+
+	return TRUE
+
+// This handles the pressure sensor hud element. Values based on human values.
+/mob/living/silicon/robot/proc/handle_pressure_damage(datum/gas_mixture/environment)
+	//by the power of Polymorph and Errorage
+	var/localpressure = environment.return_pressure()
+	var/adjusted_pressure = localpressure - ONE_ATMOSPHERE //REAL pressure
+	if(localpressure)
+		if(adjusted_pressure >= HAZARD_HIGH_PRESSURE)
+			throw_alert(SCREEN_ALARM_PRESSURE, /obj/abstract/screen/alert/robot/pressure/high, 2)
+		else if(localpressure >= WARNING_HIGH_PRESSURE && localpressure < WARNING_HIGH_PRESSURE)
+			throw_alert(SCREEN_ALARM_PRESSURE, /obj/abstract/screen/alert/robot/pressure/high, 1)
+		else if(localpressure <= WARNING_LOW_PRESSURE && localpressure > HAZARD_LOW_PRESSURE)
+			throw_alert(SCREEN_ALARM_PRESSURE, /obj/abstract/screen/alert/robot/pressure/low, -1)
+		else if(localpressure <= HAZARD_LOW_PRESSURE)
+			throw_alert(SCREEN_ALARM_PRESSURE, /obj/abstract/screen/alert/robot/pressure/low, -2)
+		else
+			clear_alert(SCREEN_ALARM_PRESSURE)
+	else //there ain't no air, we're in a vacuum
+		throw_alert(SCREEN_ALARM_PRESSURE, /obj/abstract/screen/alert/robot/pressure/low, -2)
+
+// This handles the temp sensor hud element
+/mob/living/silicon/robot/proc/handle_heat_damage(datum/gas_mixture/environment)
+	var/envirotemp = environment.return_temperature()
+	if(environment)
+		if(envirotemp)
+			if (envirotemp >= 1000 ) //1000 is the heat_level_3 for humans
+				throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/hot, 2)
+			else if (envirotemp >= BODYTEMP_HEAT_DAMAGE_LIMIT && envirotemp < 1000 )
+				throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/hot, 1)
+			else if (envirotemp <= T0C && envirotemp > BODYTEMP_COLD_DAMAGE_LIMIT)
+				throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/cold, -1)
+			else if (envirotemp <= BODYTEMP_COLD_DAMAGE_LIMIT ) //space is cold
+				throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/cold, -2)
+			else
+				clear_alert(SCREEN_ALARM_TEMPERATURE)
+				return FALSE
+	else //vacuums are cold
+		throw_alert(SCREEN_ALARM_TEMPERATURE, /obj/abstract/screen/alert/robot/temp/cold, -2)
+
+/mob/living/silicon/robot/proc/handle_health_hud()
+	if(healths)
+		if(!isDead())
+			switch(health)
+				if(200 to INFINITY)
```

Comment:
Fairly sure mommis dont have 200hp

Shouldnt this be max_health, and go down by a fifth each?